### DISCUSSION TITLE
Control Center update vs Terminal

### DISCUSSION THREAD

**Discussion by ghost** [linuxdeepin/developer-center#3178] [Q&A | 问答板块] [answered]
Hi, 
I have a question, is it better to check for updates from the Terminal (sudo apt update && upgrade) or opening settings and use the control center integrated option?🤔

**Comment by NjlsShade** [comment]
From my understanding, the Control Center will only show updates available when an OS update has been pushed directly by deepin, whereas using apt manually via the terminal will provide you with package updates as soon as they are available on the repository.

If you don't need updated packages as soon as they are released then using the Control Center is the safer and intended option, as it's more curated and also creates a backup before applying updates if you installed using default partitioning.

**Comment by BLumia** [comment, marked as answer]
The Control Center will only check updates with the system official package repos (i.e. the `/etc/apt/sources.list` file only, files under the `sources.list.d` folder doesn't count), if you installed/configured 3rd-party repo, related package updates will not display there (ps. 3rd-party repo can be showed here if you enable the option in "Update Settings" tab), it's be done by passing specific `apt` arguments (I haven't dug into the source code yet so not sure what's exactly the arguments).

Using `apt update && apt upgrade` manually, of course, will use its default behavior which includes all your configured package repos :)

**Reply by NjlsShade** [reply]
> 

Is the Control Center supposed to show official package repo updates as soon as they are available? From my experience it seems to only show updates when deepin has a public major or minor os update.

The screenshot below shows that there are updates available for applications from the official repo, however, neither the Control Center or the App Store show these as updatable.

![Screenshot_20220523232026](https://user-images.githubusercontent.com/15788214/169943023-f6036005-c82c-4047-a223-6f66c737b2b0.png)

**Reply by BLumia** [reply]
> Is the Control Center supposed to show official package repo updates as soon as they are available?

I edited my answer so it can be better describing its behavior. The control center update page will only shows the updates in the repos listed in the `/etc/apt/sources.list` file which only contains the core components of the OS, other first party packages are under the `sources.list.d` folder. For example, the calendar app is in `sources.list.d/appstore.list`. And you guess it, the App Store will show package updates using the `appstore.list` source list :)

Additionally, the store daemon only pulls for update every few hours, so it might not show updates immediately, if you wait for a while, the AppStore will show the update you want :)

**Comment by ghost** [comment]
I am usually using terminal for updates and it is covering both the system updates from the control center and third party apps installed by me. So terminal for wider type of packages, control center for Deepin system updates
